modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/resources/attachment/Attachment.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.kg.core.resources.attachment
+
+import ch.epfl.bluebrain.nexus.kg.core.resources.attachment.Attachment.Info
+
+/**
+  * Holds all the metadata information related to an attachment.
+  *
+  * @param fileUri uri where the attachment gets stored. TODO: Change the type to a refined type
+  * @param info    extra information about the attachment
+  */
+final case class Attachment(fileUri: String, info: Info) {
+  val name: String = info.originalFileName
+}
+
+object Attachment {
+
+// TODO: Use refined type for fileUri
+//    final def apply(uri: String, info: Info): Either[String, Attachment] =
+//      if (uri == null || uri.isEmpty) Left("'fileUri' field cannot be empty")
+//      else Right(new Attachment(uri, info))
+
+  /**
+    * Holds all metadata information related to an attachment
+    * that we want to expose through the service.
+    *
+    * @param originalFileName the original filename of the attached file
```

Comment:
if we support multiple attachments, we can now consider using `filename` instead of `originalFileName`.